dot/digest/digest.go

```diff
@@ -225,6 +257,35 @@ func (h *Handler) handleBlockFinalisation(ctx context.Context) {
 	}
 }
 
+// persistBABEDigestsForNextEpoch is called only when a block is finalised
+// and defines the correct next epoch data and next config data.
+func (h *Handler) persistBABEDigestsForNextEpoch(finalizedHeader *types.Header) error {
+	currEpoch, err := h.epochState.GetEpochForBlock(finalizedHeader)
+	if err != nil {
+		return fmt.Errorf("cannot get epoch for block %d (%s): %w",
+			finalizedHeader.Number, finalizedHeader.Hash(), err)
+	}
+
+	nextEpoch := currEpoch + 1
+	err = h.epochState.FinalizeBABENextEpochData(nextEpoch)
+	if err != nil && !errors.Is(err, state.ErrEpochNotInMemory) {
+		return fmt.Errorf("cannot finalize babe next epoch data for block number %d (%s): %w",
+			finalizedHeader.Number, finalizedHeader.Hash(), err)
+	}
+
+	err = h.epochState.FinalizeBABENextConfigData(nextEpoch)
+	if err == nil {
+		return nil
+	} else if errors.Is(err, state.ErrEpochNotInMemory) {
+		return ErrDefineNextEpoch
```

Comment:
nit should we also add the detail error as a string?

```suggestion
		return fmt.Errorf("%w: %s", ErrDefineNextEpoch, err)
```